src/ggrc/assets/javascripts/components/csv/export.js

```
@@ -0,0 +1,219 @@
+        type: url.model_type || "Program",
+        editFilename: false,
+        only_relevant: false,
+        filename: "Export Objects",
+        data_grid: function () {
+          return _.has(url, "data_grid");
+        }
+      });
+
+  can.Component.extend({
+    tag: "csv-export",
+    template: "<content></content>",
+    scope: function () {
+      return {
+        export: new exportModel()
+      };
+    },
+    events: {
+      ".report-title-trigger click": function (el, ev) {
+        ev.preventDefault();
+        this.scope.attr("export.editFilename", true);
+      },
+      ".report-title-change .btn-primary click": function (el, ev) {
+        ev.preventDefault();
+        this.scope.attr("export.editFilename", false);
+      },
+      ".save-template .btn-success click": function (el, ev) {
+        ev.preventDefault();
+
+        var panels = this.scope.attr("export.panels.items");
+        query = _.map(panels, function(panel, index){
+          return {
+            object_name: panel.type,
+            fields: _.compact(_.map(Object.keys(panel.selected), function(key){
+              return panel.selected[key] === true ? key : null;} )),
+            filters: {
+              relevant_filters: [_.map(panel.relevance(), function(el){
+                return {object_name: el.model_name, ids: [el.filter.id]}
+              })],
+              object_filters: GGRC.query_parser.parse(panel.filter || "")
+            }
+          }
+        });
+
+        $.ajax({
+          type: "POST",
+          dataType: "text",
+          headers: {
+            "Content-Type": "application/json",
+            "X-requested-by": "gGRC",
+            "X-export-view": "blocks"
+          },
+          url: this.scope.attr("export.url"),
+          data: JSON.stringify(query)
+        }).then(function(data){
+          GGRC.Utils.download("hello.csv", data);
+        }.bind(this))
+        .fail(function(data){
+          // TODO: report error
```

Can you show a warning flash is something goes wrong instead of failing silently? Or at least `console.warn('something')`